src/imp/test/pidinfo.c

```diff
@@ -11,12 +11,140 @@
 #include <stdio.h>
 #include <errno.h>
 #include <sys/types.h>
+#include <wait.h>
 #include <unistd.h>
+#include <string.h>
 
 #include "pidinfo.h"
 
 #include "src/libtap/tap.h"
 
+static pid_t testchild_create (int nchildren)
+{
+    int pfd[2];
+    pid_t pid;
+    char c;
+
+    if (pipe (pfd) < 0)
+        BAIL_OUT ("pipe: %s", strerror (errno));
+
+    pid = fork ();
+    if (pid < 0)
+        return -1;
+    if (pid == 0) {
+        pid_t cpid[nchildren];
+
+        /* Child must also have some children */
+        int status;
+        int exitcode = 4;
+
+        /*  Close parent end of pipe */
+        close (pfd[0]);
+
+        /*  Close stderr/out to avoid confusing libtap */
+        close (STDERR_FILENO);
+        close (STDOUT_FILENO);
+
+        for (int i = 0; i < nchildren; i++) {
+            cpid[i] = fork ();
+            if (cpid[i] < 0)
+                exit (2);
+            if (cpid[i] == 0) {
+                close (pfd[1]);
+                pause ();
+                exit (1);
+            }
+
+        }
+        alarm (5);
+        /*  Close child end of pipe */
+        close (pfd[1]);
+        for (int i = 0; i < nchildren; i++) {
+            int code = -1;
+            if (waitpid (cpid[i], &status, 0) < 0)
+                exit (3);
+            if (WIFEXITED (status))
+                code = WEXITSTATUS (status);
+            else if (WIFSIGNALED (status))
+                code = WTERMSIG (status) + 128;
+            if (code > exitcode)
+                exitcode = code;
+        }
+        exit (exitcode);
+    }
+    close (pfd[1]);
+
+    /* block until children close pipe */
+    if (read (pfd[0], &c, 1) != 0)
+        BAIL_OUT ("read from child pipe failed: %s", strerror (errno));
+    close (pfd[0]);
+    return pid;
+}
+
+static void pid_kill_tests (void)
+{
+    pid_t pid;
+    int status;
+
+    ok (pid_kill_children (-1, 0) < 0 && errno == EINVAL,
+        "pid_kill_children with invalid args returns EINVAL: got %d", errno);
```

Comment:
errno needs to be cleared before tests that are verifying that it's been set, otherwise the result from the previous test can cause a false positive.